src/tests/examples/antares_test_utils.py

```diff
@@ -90,46 +58,132 @@ def enable_study_output(study_path, enable):
+        assert ref_column_titles.issubset(other_column_titles),\
+               f"The following column(s) is missing in the reference {ref_column_titles.difference(other_column_titles)}"
+
+        for col_name in ref_column_titles:
+            try:
+                pd.testing.assert_series_equal(ref_data_frame[col_name], other_data_frame[col_name], \
+                                               atol=tol.absolute(col_name), rtol=tol.relative(col_name))
+            except AssertionError:  # Catch and re-raise exception to print col_name & tolerances
+                # Reduced path of files we're comparing
+                file_path = sep.join(file_pair[REF_INDEX].absolute().parts[-4:])
+
+                print_comparison_report(ref_data_frame, other_data_frame, file_path, col_name)
+                at_least_one_diff = True
+
+    return not at_least_one_diff
+
+class Tolerances():
+    RTOL = {}
+    ATOL = {}
+    default_abs_tol = 0
+    default_rel_tol = 1e-4
+
+    def absolute(self, col_name):
+        trimmed_col_name = trim_digit_after_last_dot(col_name)
+        if trimmed_col_name in self.ATOL:
+            return self.ATOL[trimmed_col_name]
+        return self.default_abs_tol
+
+    def relative(self, col_name):
+        trimmed_col_name = trim_digit_after_last_dot(col_name)
+        if trimmed_col_name in self.RTOL:
+            return self.RTOL[trimmed_col_name]
+        return self.default_rel_tol
+
+    def set_relative(self, name, value):
+        self.RTOL[name] = value
+
+    def set_absolute(self, name, value):
+        self.ATOL[name] = value
+
+class Linux_tolerances(Tolerances):
+    def __init__(self):
+        self.RTOL = {"CO2 EMIS.": 1e-3, "FLOW LIN.": 1e-3, "UCAP LIN.": 1e-3, "H. INFL": 1e-3, "H. STOR": 1e-3,
+                           "H. OVFL": 1e-3, "OV. COST": 1e-3, "LIGNITE": 1e-3, "CONG. FEE (ABS.)": 1e-3, "sb": 1e-3,
+                           "MARG. COST": 1e-3, "DTG MRG": 1e-3, "BALANCE": 1e-3, "BASE": 1e-3, "MRG. PRICE": 1e-3,
+                           "OP. COST": 1e-3, "SEMI BASE": 1e-3, "COAL": 1e-3, "MAX MRG": 1e-3, "UNSP. ENRG": 1e-3}
+        self.ATOL = {"CO2 EMIS.": 1, "CONG. FEE (ALG.)": 1, "FLOW LIN.": 1, "UCAP LIN.": 1, "peak": 1, "PEAK": 1,
+                           "H. INFL": 1, "H. STOR": 1, "HURDLE COST": 1, "H. OVFL": 1, "LOAD": 1, "CONG. FEE (ABS.)": 1,
+                           "sb": 1, "MISC. DTG": 1, "DTG MRG": 1, "BALANCE": 1, "BASE": 1, "OP. COST": 1,
+                           "SEMI BASE": 1, "COAL": 1, "p": 1, "MAX MRG": 1, "UNSP. ENRG": 1, "SOLAR": 1, "b": 1,
+                           "NODU": 1, "H. ROR": 1}
+
+class Win_tolerances(Tolerances):
+    def __init__(self):
+        pass
+
+
+def get_tolerances():
+    if sys.platform == "linux":
+        return Linux_tolerances()
+    elif sys.platform == "win32":
+        return Win_tolerances()
+    else:
+        print("Unknown OS")
+        raise AssertionError()
```

Comment:
raise AssertionError("Unknown OS")

Reply:
Indeed, that would be better. But option **--tb=no** when calling **output_compare_test.py** prevents any trace back report.
We could use  **--tb=line** instead (one line of trace back per failure).
But in that case we have a report like this (2 tests are failing) : 
```
================================== FAILURES ===================================
D:\Users\pierregui\Desktop\ant-open-rel\src\tests\examples\antares_test_utils.py:178: AssertionError: Comparison failed
D:\Users\pierregui\Desktop\ant-open-rel\src\tests\examples\antares_test_utils.py:178: AssertionError: Comparison failed
=========================== short test summary info ===========================
FAILED output_compare_test.py::test_065_pumped_storage_plant_explicit_model_01
FAILED output_compare_test.py::test_067_pumped_storage_plant_explicit_model_03
================= 2 failed, 1 passed, 82 deselected in 34.11s =================
```
We have one trace back per failure, but failed assertion are not associated to a test. So we could insert the test name in the assertion message.